apps/backend/src/app/modules/users/users.service.ts

```diff
@@ -747,5 +775,38 @@ export class UsersService {
     return new PagedResponseDto(MapCreditDto, dbResponse);
   }
 
+  async findSteamFriends(userID: number, steamID: bigint): Promise<UserDto[]> {
+    const steamFriends = await this.steamService.getSteamFriends(steamID);
+    const steamIDs = steamFriends.map((friend) => BigInt(friend.steamid));
+
+    const users = await this.db.user.findMany({
+      where: {
+        steamID: { in: steamIDs }
+      }
+    });
+
+    if (users.length === 0) return [];
+
+    const userIDs = users.map((user) => user.id);
+    const followerRelations = await this.db.follow.findMany({
+      where: {
+        followeeID: userID,
+        followedID: { in: userIDs }
+      }
+    });
+    const followingSet = new Set(
+      followerRelations.map((rel) => rel.followedID)
+    );
+
+    const steamIDSet = new Set(steamIDs.map((id) => id));
+    const filteredUsers = users
+      .filter(
+        (user) => steamIDSet.has(user.steamID) && !followingSet.has(user.id)
+      )
+      .map((user) => DtoFactory(UserDto, user));
```

Comment:
I think this whole query can be just
```ts
    this.db.user.findMany({
      where: {
        steamID: { in: steamIDs },
        follows: { none: { followeeID: userID } }
      }
    });
```
though I'm half asleep atm so maybe that's wrong.